File: MultiLanguageSandbox/src/test/deps/elixir/12.exs
defmodule SearchInsertPosition do
def find_insert_position(nums, target) do
    binary_search(nums, target, 0, length(nums) - 1)
  end

  defp binary_search(nums, target, left, right) when left > right do
    left
  end

  defp binary_search(nums, target, left, right) do
    mid = div(left + right, 2)
    case Enum.at(nums, mid) do
      val when val == target -> mid
      val when val < target -> binary_search(nums, target, mid + 1, right)
      _ -> binary_search(nums, target, left, mid - 1)
    end
  end
end



ExUnit.start()
defmodule SearchInsertPositionTest do
use ExUnit.Case

test "find insert position" do
assert SearchInsertPosition.find_insert_position([1, 3, 5, 6], 5) == 2
assert SearchInsertPosition.find_insert_position([1, 3, 5, 6], 2) == 1
assert SearchInsertPosition.find_insert_position([1, 3, 5, 6], 7) == 4
assert SearchInsertPosition.find_insert_position([1, 3, 5, 6], 0) == 0
end
end

File: MultiLanguageSandbox/src/test/deps/swift/12.swift
func mostFrequentTargetAfterKey(_ nums: [Int], _ key: Int) -> Int {
    var frequencyDict = [Int: Int]()
    
    for i in 0..<nums.count - 1 {
        if nums[i] == key {
            let target = nums[i + 1]
            frequencyDict[target] = (frequencyDict[target] ?? 0) + 1
        }
    }
    
    // Find the target with the highest frequency
    var mostFrequentTarget = 0
    var maxFrequency = 0
    
    for (target, frequency) in frequencyDict {
        if frequency > maxFrequency {
            maxFrequency = frequency
            mostFrequentTarget = target
        }
    }
    
    return mostFrequentTarget
}


func checkMostFrequentTargetAfterKey(_ mostFrequentTargetAfterKey: ([Int], Int) -> Int) {
assert(mostFrequentTargetAfterKey([1, 2, 1, 2, 3, 1, 2, 3], 1) == 2)
assert(mostFrequentTargetAfterKey([1, 1, 1, 2, 2, 2, 1, 3, 3], 1) == 1)
assert(mostFrequentTargetAfterKey([1, 2, 3, 1, 2, 3, 1, 2], 2) == 3)
assert(mostFrequentTargetAfterKey([7, 7, 7, 8, 8, 9, 7, 9, 9], 7) == 7)
assert(mostFrequentTargetAfterKey([1], 1) == 1)
assert(mostFrequentTargetAfterKey([2, 2, 2, 2, 2, 3], 2) == 2)
}

checkMostFrequentTargetAfterKey(mostFrequentTargetAfterKey)